modules/scaleform.lua

```diff
@@ -0,0 +1,208 @@
+function Scaleform:SetProperties(x, y, width, height)
+    if self.fullScreen then -- If the full screen is true
+        return error('Cannot set properties when full screen is enabled') -- Error cannot set properties when full screen is enabled
+    end
+    self.x = x
+    self.y = y
+    self.width = width
+    self.height = height
+end
+
+
+---@param name string
+---@param model string|number
+---@return nil
+---@description Create a render target for the scaleform - optional , only if you want to render the scaleform in 3D
+function Scaleform:RenderTarget(name, model)
+    if type(model) == 'string' then -- If the type of model is a string
+        model = joaat(model) -- Convert the model to a hash
+    end
+
+    if not IsNamedRendertargetRegistered(name) then -- If the named render target is not registered
+        RegisterNamedRendertarget(name, false) -- Register the named render target
+
+        if not IsNamedRendertargetLinked(model) then -- If the named render target is not linked
+            LinkNamedRendertarget(model) -- Link the named render target
+        end
+
+        self.renderTarget = GetNamedRendertargetRenderId(name) -- Get the named render target render id
+        self.targetName = name -- Set the target name
+    end
+end
+
+---@param shouldDraw boolean
+---@return nil
+---@description Draw the scaleform
+function Scaleform:Draw(shouldDraw)
+    if self.draw == shouldDraw then -- If the draw is equal to should draw
+        return -- Return
+    end
+
+    self.draw = shouldDraw -- Set the draw to should draw
+    if shouldDraw then -- If should draw is true
+
+       CreateThread(function()  -- Create a thread
+            while self.draw do -- While the draw is true
+
+                if self.renderTarget then -- If the render target is true
+                    SetTextRenderId(self.renderTarget) -- Set the text render id
+                    SetScriptGfxDrawOrder(4) -- Set the script gfx draw order
+                    SetScriptGfxDrawBehindPausemenu(true) -- allow it to draw behind pause menu
+                    SetScaleformFitRendertarget(self.handle, true)
+                end
+
+                if self.fullScreen then
+                    DrawScaleformMovieFullscreen(self.handle, 255, 255, 255, 255, 0)
+                else
+                    if not self.x or not self.y or not self.width or not self.height then
+                        error('Properties not set for scaleform') -- Error properties not set for scaleform
+                        DrawScaleformMovieFullscreen(self.handle, 255, 255, 255, 255, 0)
+                    else
+                        DrawScaleformMovie(self.handle, self.x, self.y, self.width, self.height, 255, 255, 255, 255, 0)
+                    end
+                end
+
+                if self.renderTarget then -- If the render target is true
+                    SetTextRenderId(1) -- Reset the text render id
+                end
```

Comment:
Is this required for stuff rendered after this scaleform? It would be nice if the comments explained the purpose rather than what every single line is doing, when the code is fairly readable already.

Reply:
yes, otherwise things afterwards would be trying to render upon the current target, so, it sets the Id, draws what it needs to and then changes it back